src/modules/fundingManager/bondingCurveFundingManager/BondingCurveFundingManagerBase.sol

```diff
@@ -0,0 +1,270 @@
+            tradeFeeCollected += feeAmount;
+        }
+        // Calculate mint amount based on upstream formula
+        mintAmount = _issueTokensFormulaWrapper(_depositAmount);
+        // Revert when the mint amount is lower than minimum amount the user expects
+        if (mintAmount < _minAmountOut) {
+            revert BondingCurveFundingManagerBase__InsufficientOutputAmount();
+        }
+        // Mint tokens to address
+        _mint(_receiver, mintAmount);
+        // Emit event
+        emit TokensBought(_receiver, _depositAmount, mintAmount, _msgSender());
+    }
+
+    /// @dev Opens the buy functionality by setting the state variable `buyIsOpen` to true.
+    function _openBuy() internal {
+        if (buyIsOpen == true) {
+            revert BondingCurveFundingManager__BuyingAlreadyOpen();
+        }
+        buyIsOpen = true;
+        emit BuyingEnabled();
+    }
+
+    /// @dev Closes the buy functionality by setting the state variable `buyIsOpen` to false.
+    function _closeBuy() internal {
+        if (buyIsOpen == false) {
+            revert BondingCurveFundingManager__BuyingAlreadyClosed();
+        }
+        buyIsOpen = false;
+        emit BuyingDisabled();
+    }
+
+    /// @dev Sets the buy transaction fee, expressed in BPS.
+    /// @param _fee The fee percentage to set for buy transactions.
+    function _setBuyFee(uint _fee) internal {
+        if (_fee >= BPS) {
+            revert BondingCurveFundingManager__InvalidFeePercentage();
+        }
+        emit BuyFeeUpdated(_fee, buyFee);
+        buyFee = _fee;
+    }
+
+    /// @dev Calculates the net amount after fee deduction and the fee amount based on
+    /// a transaction amount and a specified fee percentage.
+    /// @param _transactionAmount The amount involved in the transaction before fee deduction.
+    /// @param _feePct The fee percentage to be deducted, represented in basis points (BPS).
+    /// @return netAmount The transaction amount after fee deduction.
+    /// @return feeAmount The amount of fee deducted from the transaction amount.
+    function _calculateNetAmountAndFee(uint _transactionAmount, uint _feePct)
+        internal
+        pure
+        returns (uint netAmount, uint feeAmount)
+    {
+        // Calculate fee amount
+        feeAmount = (_transactionAmount * _feePct) / BPS;
+        // Calculate net amount after fee deduction
+        netAmount = _transactionAmount - feeAmount;
+    }
+
+    /// @dev Sets the number of decimals for the token.
+    /// This function updates the `tokenDecimals` state variable and should be be overriden by
+    /// the implementation contract if input validation is needed.
+    /// @param _decimals The number of decimals to set for the token.
+    function _setTokenDecimals(uint8 _decimals) internal virtual {
+        uint8 oldDecimals = tokenDecimals;
+        tokenDecimals = _decimals;
+        emit TokenDecimalsUpdated(oldDecimals, tokenDecimals);
+    }
```

Comment:
I would disagree. As the Natspec comment highlights, this function should be overridden in case of input validation, which is what we do. The BondingCurveFundingMangerBase abstract contract should be seen as a building block, for future BC implementations, which might not need the input validation and can just use the function you've highlighted